### PR TITLE
added Priority Queue

### PR DESCRIPTION
**Pull request by Bhupesh-V**
**Fixes issue:**

**Changes:**

-  Added C++ implementation of *Priority Queues*.
- Added corresponding `README.md` file explaining  *Priority Queues*.

### CODE REVIEW

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#include <cstdlib>
+ 
+// Node Declaration
+struct node{
+	int priority;
+	int info;
+	struct node *link;
+};
+
+class Priority_Queue{
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: Don't use underscore.

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    private:
+        node *front;
+    public:
+        Priority_Queue(){
+            front = NULL;
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: Use `nullptr`.

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+#include <iostream>
+#include <cstdio>
+#include <cstring>
+#include <cstdlib>
+ 
+// Node Declaration
+struct node{
+	int priority;
+	int info;
+	struct node *link;
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: In C++, the `struct node*` isn't necessary.

**Comment by Bhupesh-V** [comment]
Changes done

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 4]
```diff
@@ -9,29 +9,29 @@
 struct node{
 	int priority;
 	int info;
-	struct node *link;
+	node *link;
 };
 
-class Priority_Queue{
+class PriorityQueue{
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: Fix coding style (spaces before braces).

Comment (Bhupesh-V) [reply code/data_structures/src/queue/priority_queue/priority_queue.cpp]: fixed

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        case 3:
+            pq.display();
+            break;
+        case 4:
+            break;
+        default :
+            std::cout << "Wrong Choice !!!\n";
+        }
+    }
+    while(choice != 4);
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: Move this to previous line.

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    PriorityQueue pq; 
+
+    do{
+        std::cout << "1.Insert\n";
+        std::cout << "2.Delete\n";
+        std::cout << "3.Display\n";
+        std::cout << "4.Quit\n";
+        std::cout << "Enter your choice : "; 
+        std::cin >> choice;
+        switch(choice){
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]:
```suggestion
        switch(choice) {
```

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+ 
+// Node Declaration
+struct node {
+	int priority;
+	int info;
+	node *link;
+};
+
+class PriorityQueue {
+    private:
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: Unindent the whole class body.

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 8]
```diff
@@ -0,0 +1,107 @@
+// C++ implementation of Priority Queue
+
+#include <iostream>
+#include <cstdio>
+#include <cstring>
+#include <cstdlib>
+ 
+// Node Declaration
+struct node {
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: Call this `Node`.

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+                tmp->link = q->link;
+                q->link = tmp;
+            }
+        }
+        
+        // Delete from Priority Queue
+        void remove() {
+            node *tmp;
+            if(front == nullptr)
+                std::cout << "Queue Underflow\n";
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: Don't print anything out; raise an exception or return a boolean instead; handle errors in `main`.

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+            else {
+                tmp = front;
+                std::cout << "Removed item is: " << tmp->info << "\n";
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: Remove.

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp (hunk 5 above)
Comment (Bhupesh-V) [reply code/data_structures/src/queue/priority_queue/priority_queue.cpp]: done

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp (hunk 6 above)
Comment (Bhupesh-V) [reply code/data_structures/src/queue/priority_queue/priority_queue.cpp]: done

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp (hunk 7 above)
Comment (Bhupesh-V) [reply code/data_structures/src/queue/priority_queue/priority_queue.cpp]: done

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp (hunk 9 above)
Comment (Bhupesh-V) [reply code/data_structures/src/queue/priority_queue/priority_queue.cpp]: done

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+    }
+    
+    // Removes an element with highest priority from Priority Queue 
+    void remove() {
+        node *tmp;
+        if(front == nullptr)
+            throw true;
+        else {
+            tmp = front;
+            std::cout << "Removed item is: " << tmp->info << "\n";
```
Comment (arnavb) [inline code/data_structures/src/queue/priority_queue/priority_queue.cpp]: Remove this.

**Comment by Vaibhavi24** [comment]
@Bhupesh-V waiting for your changes!

**Comment by Bhupesh-V** [comment]
> @Bhupesh-V waiting for your changes!

Sorry I have been busy with college internals :sweat_smile:

File: code/data_structures/src/queue/priority_queue/priority_queue.cpp (hunk 11 above)
Comment (Bhupesh-V) [reply code/data_structures/src/queue/priority_queue/priority_queue.cpp]: I am actually not clear why should I remove this ??

**Pull request closed without merging**